WcaOnRails/app/models/competition.rb

```diff
@@ -266,11 +294,37 @@ def registration_must_close_after_it_opens
     end
   end
 
+  def use_wca_as_website=(value)
+    @use_wca_as_website = ActiveRecord::Type::Boolean.new.type_cast_from_database(value)
+  end
+
+  def use_wca_as_website
+    @use_wca_as_website || website == url
+  end
+
+  before_validation :assign_wca_website_url
+  private def assign_wca_website_url
+    if use_wca_as_website && website != url
+      self.website = self.url
+    end
+  end
+
+  after_save :update_website
+  private def update_website
+    if use_wca_as_website && website != url
+      update_column :website, url
```

Comment:
The big problem here is that I simply don't understand what this code does. We need readable and understandable code.

Trying to figure it out:
- There is a db attribute `website`, which I think is the pre-existing external web site for the competition.
- There is a kinda emulated non DB boolean attribute `use_wca_as_website`
- There is the 'url' function, which I think is the URL for what I'll call the "internal" site.

I think the idea is to use `.website` for both the external and internal url. This is complicated by the fact that the internal URL can change if the `id` changes, so we update it, just in case, after each save. I can't quite make out what's going on with `use_wca_as_website` the function vs the @-var.

This may well work, and if it doesn't, it can be made to. But I think a straightforward 'standard' solution that people can easily understand and work with is much preferable.

I would rename `use_wca_as_website`  to `generate_website`, and make it a DB column. Then I _think_ these 2 one-liners takes care of everything:

    def website
       generate_website ? generated_url : external_website
    end

   def generated_url
     Rails.application.routes.url_helpers.competition_url(self, host: ENVied.ROOT_URL)
   end